### PR TITLE
Add option for 'ON CONFLICT DO UPDATE'

### PR DESCRIPTION
**Pull request by poscat0x04**

### CODE REVIEW

**Comment by tomjaguarpaw** [comment]
Thanks! I never used this feature though so I'm unsure about how it is supposed to be used.  [The `INSERT` docs](https://www.postgresql.org/docs/current/sql-insert.html) suggest that `DO UPDATE` should always be followed by `SET` of some column names. Is it possible to have `DO UPDATE` on its own?

File: src/Opaleye/Internal/HaskellDB/Sql.hs [hunk 1]
```diff
@@ -65,6 +65,7 @@ data SqlUpdate  = SqlUpdate SqlTable [(SqlColumn,SqlExpr)] [SqlExpr]
 data SqlDelete  = SqlDelete SqlTable [SqlExpr]
 
 data OnConflict = DoNothing
+                | DoUpdate
                 -- ^ @ON CONFLICT DO NOTHING@
```
Comment (tomjaguarpaw) [inline src/Opaleye/Internal/HaskellDB/Sql.hs:69]: The comment has become attached to the wrong constructor.

**Comment by poscat0x04** [comment]
oops. That was dumb. I should've read the manual before making the pull request. Sorry for the trouble I've caused.

**Comment by tomjaguarpaw** [comment]
That's no problem at all.

**Pull request closed without merging**